lib/generator.js

```diff
@@ -692,6 +698,38 @@ class Generator {
 
     return Promise.all(promises);
   }
+
+  /**
+   * Check if given directory is a git repo with unstaged changes or is not empty
+   * @private
+   * @param  {String} [dir] Directory that needs to be tested for a given condition.
+  */
+  async verifyTargetDir(dir) {
+    const gitInfo = git(dir);
+
+    try {
+      const isGitRepo = await gitInfo.checkIsRepo();
+
+      if (isGitRepo) {
+        const gitStatus = await gitInfo.status();
+
+        //New files are not tracked and not visible as modified
+        const hasUntrackedUnstagedFiles = gitStatus.not_added.length !== 0;
+
+        const stagedFiles = gitStatus.staged;
+        const modifiedFiles = gitStatus.modified;
+        const hasModifiedUstagedFiles = (modifiedFiles.filter(e => stagedFiles.indexOf(e) === -1)).length !== 0;
+
+        if (hasModifiedUstagedFiles || hasUntrackedUnstagedFiles) throw new Error(`"${this.targetDir}" is in a git repository with unstaged changes`);
+      } else {
+        const isDirEmpty = (await readDir(dir)).length === 0;
+
+        if (!isDirEmpty) throw new Error(`"${this.targetDir}" is not an empty directory. You might override your work.`);
```

Comment:
We could improve the message here, with something like "To skip this rule, please make your code a git repository or use the --force-write flag (not recommended).".

Not sure "rule" is the right word here and above tho.

Reply:
`rule` sounds good to me